### PR TITLE
Clean up how we build the OpenAPI spec.

### PR DESCRIPTION
**Pull request by adk-swisstopo**
This should make it easier to inject things we need to use AWS API Gateway. It should also be a bit more robust and easier to use for external developers.

### CODE REVIEW

**Comment by adk-swisstopo** [comment]
I think the missing tag descriptions are because they are missing from https://github.com/geoadmin/service-stac/blob/develop/spec/transaction/tags.yaml
And the missing "Swissregio" value might be because this "title" field is duplicated: https://github.com/geoadmin/service-stac/blob/develop/spec/components/schemas.yaml#L1433
Unclear whether we can make yq 4 behave the same than 3 or if it's preferable to "fix" the source.

**Comment by adk-swisstopo** [comment]
Reading https://yaml.org/spec/1.2.2/#mapping I believe the following is a bug:
https://github.com/geoadmin/service-stac/blob/27eef5cbd44a827f017a68d293aa1694a2b37db6/spec/components/schemas.yaml#L1433
```
      example:
        title:
          eq: "Swissregio"
        title:
          contains: "Swiss"
```
It defines a "Mapping" with two keys "title". Within a mapping, each key needs to be unique. So, yq 4 keeping just one of the keys is entirely reasonable.

File: spec/Makefile [hunk 1]
```diff
@@ -28,33 +31,38 @@ help:
...
-
+	$(OPENAPI_VALIDATOR) -e $(OPENAPI)
+	$(OPENAPI_VALIDATOR) -e $(OPENAPI_TRANSACTIONAL)
+
+# This can be useful to distinguish style and semantic differences when we
+# update the tools. For example if you update yq and see lot of whitespace
+# changes, you can "make static/spec/v1/openapi.props" and diff that to check
+# for change in the properties themselves.
+%.props: %.yaml
+	$(YQ) -P 'sort_keys(..)' -o=props $(^) > $(@)
```
Comment (boecklic) [inline spec/Makefile:65]: :+1: very useful!

**Review by boecklic — APPROVED** [review]
Thanks for the cleanup and improvement!

**Pull request merged**